applications/hotornot/src/hon_util.erl

```diff
@@ -20,74 +21,97 @@
 -endif.
 
 -define(MIN_PREFIX_LEN, 1). % how many chars to strip off the e164 DID
--define(BOTH_DIRECTIONS, [<<"inbound">>, <<"outbound">>]).
 
--spec candidate_rates(ne_binary()) ->
+-spec candidate_rates(ne_binary(), api_binary(), ne_binary()) ->
                              {'ok', wh_json:objects()} |
                              {'error', atom()}.
--spec candidate_rates(ne_binary(), binary()) ->
+-spec candidate_rates(ne_binary(), api_binary(), ne_binary(), binary()) ->
                              {'ok', wh_json:objects()} |
                              {'error', atom()}.
-candidate_rates(ToDID) ->
-    candidate_rates(ToDID, <<>>).
-candidate_rates(ToDID, FromDID) ->
+candidate_rates(AccountId, Direction, ToDID) ->
+    candidate_rates(AccountId, Direction, ToDID, <<>>).
+candidate_rates(AccountId, Direction, ToDID, FromDID) ->
     E164 = wnm_util:to_e164(ToDID),
-    find_candidate_rates(E164, FromDID).
+    find_candidate_rates(AccountId, Direction, E164, FromDID).
+
+find_candidate_rates(AccountId, Direction, E164, _FromDID) when byte_size(E164) > ?MIN_PREFIX_LEN ->
+    Keys = build_keys(Direction, E164),
+    RateDbs = rate_dbs(AccountId),
+    case lists:foldl(fun acc_candidate_rates/2, {'ok', Keys, 0, []}, RateDbs) of
+        {'ok', _, _, Rates} -> {'ok', Rates};
+        E -> E
+    end;
+find_candidate_rates(_, _, DID, _) ->
+    lager:debug("DID ~s is too short", [DID]),
+    {'error', 'did_too_short'}.
 
-find_candidate_rates(E164, _FromDID) when byte_size(E164) > ?MIN_PREFIX_LEN ->
-    Keys = build_keys(E164),
+build_keys(Direction, <<"+", E164/binary>>) ->
+    build_keys(Direction, E164);
+build_keys(Direction, <<D:1/binary, Rest/binary>>) ->
+    build_keys(Direction, Rest, D, [[D, Direction]]).
+
+build_keys(Direction, <<D:1/binary, Rest/binary>>, Prefix, Acc) ->
+    build_keys(Direction, Rest, <<Prefix/binary, D/binary>>, [[<<Prefix/binary, D/binary>>, Direction] | Acc]);
+build_keys(_, <<>>, _, Acc) -> Acc.
+
+-spec rate_dbs(ne_binary()) -> list().
+rate_dbs(AccountId) ->
+    AccountDb = wh_util:format_account_db(AccountId),
+    {'ok', A} = couch_mgr:open_doc(AccountDb, AccountId),
+    lists:foldl(fun acc_rate_dbs/2, [], lists:reverse([AccountId | kz_account:tree(A)]))
```

Comment:
@jamesaimonetti i would also like to see pvt_ratedeck_id to support multiple ratedecks per ratedeck db. one reseller can have several ratedeckas and apply A to a customer and B to another.